core/src/streams/redis.rs

```diff
@@ -0,0 +1,54 @@
+use log::{error, info};
+use thiserror::Error;
+use redis::{cmd, AsyncCommands};
+use redis::aio::MultiplexedConnection;
+use serde_json::Value;
+
+#[derive(Error, Debug)]
+pub enum RedisError {
+    #[error("Redis error: {0}")]
+    RedisError(#[from] redis::RedisError),
+}
+
+#[derive(Debug, Clone)]
+pub struct Redis {
+    client: redis::Client,
+}
+
+impl Redis {
+    pub fn new(url: &str) -> Self {
```

Comment:
should follow the others and be Result<Self, RedisError> instead of panic check kafka